cumulus_library/actions/builder.py

```diff
@@ -228,35 +285,87 @@ def build_study(
-) -> list:
+    prepare: bool = False,
+    data_path: pathlib.Path | None = None,
+) -> None:
     """Creates tables in the schema by iterating through the sql_config.file_names
 
     :param config: a StudyConfig object
     :param manifest: a StudyManifest object
+    :keyword db_parser: a parser for the target database
     :keyword continue_from: Name of a file to resume table creation from
-    :returns: loaded queries (for unit testing only)
+    :keyword prepare: If true, will render query instead of executing
+    :keyword data_path: If prepare is true, the path to write rendered data to
     """
+    if prepare and manifest.get_study_prefix() in ["core", "discovery", "data-metrics"]:
+        sys.exit(
+            f"{manifest.get_study_prefix()} does not support prepare mode. It must be run "
+            "directly against a target database."
+        )
     if file_list is None:
         file_list = manifest.get_file_list(continue_from)
+    if prepare:
+        data_dir = data_path / manifest.get_study_prefix()
+        files = data_dir.glob("*")
+        [file.unlink() for file in files if file.is_file()]
+    query_count = 0
     for file in file_list:
         if file.endswith(".py"):
-            _load_and_execute_builder(
+            query_count += _load_and_execute_builder(
                 config=config,
                 manifest=manifest,
                 filename=file,
                 db_parser=db_parser,
+                prepare=prepare,
+                data_path=data_path,
+                query_count=query_count,
             )
         elif file.endswith(".toml"):
-            _run_workflow(config=config, manifest=manifest, filename=file)
+            query_count += _run_workflow(
+                config=config,
+                manifest=manifest,
+                filename=file,
+                prepare=prepare,
+                data_path=data_path,
+                query_count=query_count,
+            )
         elif file.endswith(".sql"):
-            _run_raw_queries(config=config, manifest=manifest, filename=file)
+            query_count += _run_raw_queries(
+                config=config,
+                manifest=manifest,
+                filename=file,
+                prepare=prepare,
+                data_path=data_path,
+                query_count=query_count,
+            )
         else:
             raise errors.StudyManifestParsingError
+    if prepare:
+        with zipfile.ZipFile(
+            f"{data_path}/{manifest.get_study_prefix()}.zip", "w", zipfile.ZIP_DEFLATED
+        ) as z:
+            for file in data_dir.glob("*"):
+                z.write(file, file.relative_to(data_dir))
 
 
 def _run_raw_queries(
-    config: base_utils.StudyConfig, manifest: study_manifest.StudyManifest, filename: str
-):
+    config: base_utils.StudyConfig,
+    manifest: study_manifest.StudyManifest,
+    filename: str,
+    prepare: bool,
+    data_path: pathlib.Path,
+    query_count: int,
```

Comment:
nit: again, bool and int args (plus a long arg list) make me want to require some kwargs with `*`. Not really necessary, I'm just a big fan of the feature, apparently :smile:

Reply:
yeah that's fine, same comment about being lazy with internal functions.